.github/workflows/main.yml

```diff
@@ -77,4 +77,23 @@ jobs:
         with:
           branch: gh-pages
           folder: docs/build/html
-          clean-exclude: .nojekyll
\ No newline at end of file
+          clean-exclude: .nojekyll
+
+  unit_tests:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.10.x"
+      - name: install virtual display
+        run: |
+          bash .github/workflows/install_display.sh
+          echo "DISPLAY=:0" >> $GITHUB_ENV
+
+      - name: install_revolve2
+        run: ./dev_install.sh
+      - name: install pytest dependencies
```

Comment:
You can add the pytest dependencies to the dev_install script so we also have that locally. docs and codetools requirements are already in there